FireboltNETSDK/FireboltClient.cs

```diff
@@ -293,6 +298,12 @@ private async Task<T> SendAsync<T>(HttpMethod method, string uri, HttpContent? c
             }
             else
             {
+                if (needsAccessToken && response.StatusCode == HttpStatusCode.Unauthorized && retryUnauthorized)
+                {
+                    //If we need an access token and the token is invalid (401), we try to re-establish the connection once.
+                    await EstablishConnection(true);
```

Comment:
Same as above, `await EstablishConnection(forceTokenRefresh=true);` while redundant is more readable.